Source/ISHRemote/Trisoft.ISHRemote/Cmdlets/Annotation/AddIshAnnotation.cs

```diff
@@ -0,0 +1,365 @@
+    /// </code>
+    /// <para>Add annotation providing Metadata object. Full manual mode - validation of the incoming metadata is on the API side</para>
+    /// </example>
+    /// <example>
+    /// <code>
+    /// $ishSession = New-IshSession -WsBaseUrl "https://example.com/InfoShareWS/" -IshUserName "username" -IshUserPassword  "userpassword"
+    /// $metadata = Set-IshMetadataField -Name "FISHANNOTPROPOSEDCHNGTXT" -Level Annotation -Value "My proposed change text"
+    /// $ishAnnotation = Add-IshAnnotation -IshSession $ishsession `
+    ///                     -PubLogicalId "GUID-MYPUBLICATIONLOGICALID" `
+    ///                     -PubVersion "1" `
+    ///                     -LogicalId "MYCONTENTOBJECTLOGICALID" `
+    ///                     -Version "1" `
+    ///                     -Lng "en" `
+    ///                     -Type "General" `
+    ///                     -Text "My annotation text" `
+    ///                     -Status "VANNOTATIONSTATUSUNSHARED" `
+    ///                     -Category "Comment" `
+    ///                     -Address "My annotation address" `
+    ///                     -Metadata $metadata
+    /// </code>
+    /// <para>Add annotation providing required parameters and Metadata object. Cmdlet will use the latest RevisionId for the provided content object and parameter values take precedence over metadata field values</para>
+    /// </example>
+    /// <example>
+    /// <code>
+    /// $ishSession = New-IshSession -WsBaseUrl "https://example.com/InfoShareWS/" -IshUserName "username" -IshUserPassword  "userpassword"
+    /// $metadataFilter = Set-IshMetadataFilterField -Name "VERSION" -Level Version -ValueType Value -Value "1" `
+    ///                  | Set-IshMetadataFilterField -Name "DOC-LANGUAGE" -Level Lng -ValueType Value -Value "en"
+    /// $ishObject = Get-IshDocumentObj -IshSession $ishsession -LogicalId "MYCONTENTOBJECTLOGICALID" -MetadataFilter $metadataFilter
+    ///	$metadata =	Set-IshMetadataField -Name "FISHANNOTPROPOSEDCHNGTXT" -Level Annotation -Value "My proposed change text"
+    ///	$ishAnnotation = Add-IshAnnotation -IshSession $ishsession `
+    ///                     -PubLogicalId "GUID-MYPUBLICATIONLOGICALID" `
+    ///                     -PubVersion "1" `
+    ///						-IshObject $ishObject `
+    ///                     -Type $annotationType `
+    ///                     -Text "My annotation text" `
+    ///                     -Status "VANNOTATIONSTATUSUNSHARED" `
+    ///                     -Category "Comment" `
+    ///                     -Address "My annotation address" `
+    ///						-Metadata $metadata
+    /// </code>
+    /// <para>Add annotation providing IshObject of type IshDocumentObj. Cmdlet will use the latest RevisionId for the provided content object and parameter values take precedence over metadata field values</para>
+    /// </example>
+    /// <example>
```

Comment:
Can we add a sample of the most short version... so minimal fields of Annotation... like no publogical, pubversion, no topiclogical, no address...

Reply:
I thought we wanted Add-IshAnnotation cmdlet to be strict and specific, so that you can not add annotation which is then lost..I suppose the shortest version would be this:
    ///	$ishAnnotation = Add-IshAnnotation -IshSession $ishsession `
    ///                     -PubLogicalId "GUID-MYPUBLICATIONLOGICALID" `
    ///                     -PubVersion "1" `
    ///		    -IshObject $ishObject `
    ///                     -Type $annotationType `
    ///                     -Text "My annotation text" `
    ///                     -Status "VANNOTATIONSTATUSUNSHARED" `
    ///                     -Category "Comment" `
    ///                     -Address "My annotation address" 
..which is IshObject group and without Metadata parameter